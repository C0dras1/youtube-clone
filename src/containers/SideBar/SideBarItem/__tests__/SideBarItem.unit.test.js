import React from 'react';
import {shallow} from 'enzyme';
import {SideBarItem} from '../SideBarItem';

const location = {
  pathname: '/feed/trending',
};

describe('SideBarItem', () => {
  test('renders SideBarItem without path', () => {
    const wrapper = shallow(<SideBarItem location={location} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders highlighted SideBarItem', () => {
    const wrapper = shallow(
    <SideBarItem highlighted icon='fire' label='Trending' location={location} path={'/feed/trending'} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders non-highlighted SideBarItem', () => {
    const wrapper = shallow(
    <SideBarItem icon='fire' label='Trending' location={location} />
    );
    expect(wrapper).toMatchSnapshot();
  });
})