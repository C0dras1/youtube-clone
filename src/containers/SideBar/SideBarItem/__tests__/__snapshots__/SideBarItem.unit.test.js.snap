// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem renders SideBarItem without path 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="side-bar-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SideBarItem renders highlighted SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="side-bar-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem renders non-highlighted SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="side-bar-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
